packages/amplify-graphql-api-construct-tests/src/__tests__/sql-models-common.ts

```diff
@@ -19,77 +21,696 @@ export const testGraphQLAPI = async (options: {
+  expect(createTodo5Result.id).toBeDefined();
+  expect(createTodo5Result.description).toEqual('Todo #5');
+
+  const listTodosWithLimit = await listTodo(2);
+  const listTodosWithLimitResult = listTodosWithLimit.data.listTodos;
+  expect(listTodosWithLimitResult.items.length).toEqual(2);
+  expect(listTodosWithLimitResult.nextToken).toBeDefined();
+
+  const listTodosWithNextToken = await listTodo(2, listTodosWithLimitResult.nextToken);
+  const listTodosWithNextTokenResult = listTodosWithNextToken.data.listTodos;
+  expect(listTodosWithNextTokenResult.items.length).toEqual(2);
+  // expect(listTodosWithNextTokenResult.nextToken).toBeNull();
+  console.log(listTodosWithNextTokenResult.nextToken);
+
+  // Check and validate filter
+  const listTodosWithFilter = await listTodo(10, null, { description: { contains: 'Updated' } });
+  const listTodosWithFilterResult = listTodosWithFilter.data.listTodos;
+  expect(listTodosWithFilterResult.items.length).toEqual(1);
+  expect(listTodosWithFilterResult.items).toEqual(
+    expect.arrayContaining([
+      expect.objectContaining({
+        id: getUpdatedTodo2Result.id,
+        description: 'Updated Todo #2',
+      }),
+    ]),
   );
-  const emptyListResult = await graphql(
-    apiEndpoint,
-    apiKey,
-    /* GraphQL */ `
-      query LIST_TODOS {
-        listTodos {
+  expect(listTodosWithFilterResult.nextToken).toBeNull();
+
+  const listTodosWithFilter2 = await listTodo(10, null, { description: { size: { eq: 7 } } });
+  const listTodosWithFilter2Result = listTodosWithFilter2.data.listTodos;
+  expect(listTodosWithFilter2Result.items.length).toEqual(3);
+  expect(listTodosWithFilter2Result.items).toEqual(
+    expect.arrayContaining([
+      expect.objectContaining({
+        id: createTodo3Result.id,
+        description: 'Todo #3',
+      }),
+      expect.objectContaining({
+        id: createTodo4Result.id,
+        description: 'Todo #4',
+      }),
+      expect.objectContaining({
+        id: createTodo5Result.id,
+        description: 'Todo #5',
+      }),
+    ]),
+  );
+  expect(listTodosWithFilter2Result.nextToken).toBeNull();
+
+  // Check invalid CRUD operation returns generic error message
+  const createTodo6 = await createTodo('Todo #6');
+  const createTodo6Result = createTodo6.data.createTodo;
+
+  try {
+    await createTodo('Todo #7', createTodo6Result.id);
+  } catch (error) {
+    await checkGenericError(error?.message);
+  }
+
+  const invalidId = createTodo6Result.id + 'x';
+
+  try {
+    await getTodo(invalidId);
+  } catch (error) {
+    await checkGenericError(error?.message);
+  }
+
+  try {
+    await updateTodo(invalidId, 'Updated Todo #6');
+  } catch (error) {
+    await checkGenericError(error?.message);
+  }
+
+  try {
+    await deleteTodo(invalidId);
+  } catch (error) {
+    await checkGenericError(error?.message);
+  }
+
+  // Cleanup current database for next SSM Connection test
+  const deleteTodo2 = await deleteTodo(getUpdatedTodo2Result.id);
```

Comment:
1. Consider doing the cleanup in a loop rather than breaking out each thing.
2. You already assert delete behavior above--are you testing anything meaningful with the repeated assertions here?

Reply:
Test logic refactored and cleanup process extracted in a more generic function that could be reused in future tests.